elements/pl-python-variable/pl-python-variable.py

```diff
@@ -1,52 +1,119 @@
+PREFIX_DEFAULT = ""
+SUFFIX_DEFAULT = ""
 TEXT_DEFAULT = False
 SHOW_HEADER_DEFAULT = True
 SHOW_INDEX_DEFAULT = True
 SHOW_DIMENSIONS_DEFAULT = True
+SHOW_DATATYPE_DEFAULT = False
+ADD_LINE_BREAKS_DEFAULT = False
 
 
-def prepare(element_html, data):
+def prepare(element_html: str, data: pl.QuestionData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
-    pl.check_attribs(element, required_attribs=['params-name'], optional_attribs=['text', 'no-highlight', 'prefix', 'suffix', 'show-header', 'show-index', 'show-dimensions'])
+    pl.check_attribs(
+        element,
+        required_attribs=["params-name"],
+        optional_attribs=[
+            "text",
+            "no-highlight",
+            "prefix",
+            "suffix",
+            "show-header",
+            "show-index",
+            "show-dimensions",
+            "add-line-breaks",
+            "show-dtype",
+        ],
+    )
 
 
-def render(element_html, data):
+def render(element_html: str, data: pl.QuestionData) -> str:
     element = lxml.html.fragment_fromstring(element_html)
-    force_text = pl.get_boolean_attrib(element, 'text', TEXT_DEFAULT)
-    varname = pl.get_string_attrib(element, 'params-name')
-    show_header = pl.get_boolean_attrib(element, 'show-header', SHOW_HEADER_DEFAULT)
-    show_index = pl.get_boolean_attrib(element, 'show-index', SHOW_INDEX_DEFAULT)
-    show_dimensions = pl.get_boolean_attrib(element, 'show-dimensions', SHOW_DIMENSIONS_DEFAULT)
+    force_text = pl.get_boolean_attrib(element, "text", TEXT_DEFAULT)
+    varname = pl.get_string_attrib(element, "params-name")
+    show_header = pl.get_boolean_attrib(element, "show-header", SHOW_HEADER_DEFAULT)
+    show_index = pl.get_boolean_attrib(element, "show-index", SHOW_INDEX_DEFAULT)
+    show_dimensions = pl.get_boolean_attrib(
+        element, "show-dimensions", SHOW_DIMENSIONS_DEFAULT
+    )
+    show_dtype = pl.get_boolean_attrib(element, "show-dtype", SHOW_DATATYPE_DEFAULT)
+    add_line_breaks = pl.get_boolean_attrib(
+        element, "add-line-breaks", ADD_LINE_BREAKS_DEFAULT
+    )
 
-    if varname not in data['params']:
-        raise Exception('Could not find {} in params!'.format(varname))
+    if varname not in data["params"]:
+        raise ValueError(f"Could not find {varname} in params!")
 
-    var_out = data['params'][varname]
-    html = ''
-    var_type = 'text'
+    var_out = data["params"][varname]
 
     # determine the type of variable to render
-    if isinstance(var_out, dict) and '_type' in var_out:
-        if not force_text:
-            var_type = var_out['_type']
+    if isinstance(var_out, dict) and "_type" in var_out:
         var_out = pl.from_json(var_out)
 
+    html_list: List[str] = []
+
     # render the output variable
-    if var_type == 'dataframe':
-        html += var_out.to_html(header=show_header, index=show_index, classes=['pl-python-variable-table'])
+    if isinstance(var_out, pandas.DataFrame) and not force_text:
```

Comment:
The instance check probably needs to happen on `var_type` if `dataframe-v2` is included.

Reply:
As I understand it, `dataframe-v2` will be an alternative encoding for `pl.from_json`, meaning that the conversion to `var_out` will still happen first (on the lines above).

Reply:
Actually, if your main use case is using dataframes, I don't think it's unreasonable to bundle that in with these changes here. I can can close that other PR later. Though this means that this PR is going to have to wait for the BigO element to go in, since that one modifies the backend Python code too (and I don't want to have to rebase that on top of changes made here).

Reply:
@coatless Added new dataframe encoding and corresponding tests. Might be worth adding in more tests with a few more test frames but that can be done on top of the existing framework. Now resolves #6501 as well.